common/common-ui/src/main/java/com/duckduckgo/common/ui/view/dialog/CustomAlertDialogBuilder.kt

```diff
@@ -30,6 +30,7 @@ class CustomAlertDialogBuilder(val context: Context) : DaxAlertDialog {
     abstract class EventListener {
         open fun onDialogShown() {}
         open fun onDialogDismissed() {}
+        open fun onDialogCancelled() {}
```

Comment:
Android docs say about `setOnDismissListener`: "Sets the callback that will be called when the dialog is dismissed for any reason."
About `setOnCancelListener` Android docs says: "Even in a cancelable dialog, the dialog may be dismissed for reasons other than being canceled or one of the supplied choices being selected. If you are interested in listening for all cases where the dialog is dismissed and not just when it is canceled, see setOnDismissListener."

Do we need the cancel one?